### ISSUE TITLE
How do we add sourcemaps into dynamically evaluated code?

### ISSUE THREAD

**Issue by ceremcem** [gkz/LiveScript#1077]
In [my app](https://github.com/ceremcem/aecad/blob/88ce1897b80366c5ee3ab2888397c30175956286/webapps/main/sketcher/gui/scripting.ls#L36), dynamically generated code is run (just like in livescript.net) in browser. How can we add sourcemaps for those dynamically generated scripts?

**Comment by rhendric** [comment]
You can add a key called `map` to the options object passed to `compile`. The valid values are the same as for the `--map` option on the command line. `compile` will return an object instead of a string, with `code` and `map` fields. If you're passing the `code` on to something that understands `//# sourceMappingURL` directives, then using the `'embedded'` option will stuff the source map into `code`, so at that point you're done. If not, you'll want to use `'linked-src'`, and process the `map` yourself; I think it's an instance of [SourceMapGenerator](https://github.com/mozilla/source-map#sourcemapgenerator).

You may also want to add a `filename` key to your compile options, since otherwise your source map will reference `unnamed-xxxxxxxx.js`.

**Comment by ceremcem** [comment]
Thanks for your help. It turns out that this (my) problem was containing multiple issues: [Explanation](https://ractive.js.org/playground/index.html?env=docs&hidemenu#N4IgLiBcIgNCBnKID0AqABAHQHYGIMBJHMAUxIEsB7HSbHXARgDoMB1UjHU0gEwzBUMpAB6kAxgFcyGXgE8cAQwC2FcYoA2GuRnEAnUorL9xVXqQT0ATKwCiI8aQAOYajksIAFlUkbeOAHIwDAAjA0UAawwfPQxFJycMAAowT0lLAHdObj4BIXTOMD05FHUwcU8ASlwAZlYODC8fP1DORRCNQqFzEMkAc2EHZ1caSySMz1IDAWLSowqMJwMAN3IwS3S82SEEKmqGfAwAJWc9M0lxEYOmVgB5J3IMORjQs4yEKdlSXr6+z9wbBgAKo4UzKZRrDAAMwoegQwVWcLctVYR0kOAEk10Zk4uAALKwACLffp-WIZChaDCeCjBVKcUx6AyXRZUBC0tzMXAAVlYAGEqODIT5gjC4QipuyaLgAGysEFgiEkRoSGj8RFSjG4ADsqPRmIZOPoAA4iSTfp8KVSoYpKVsmhkDdimRJgk42RyaKxcBhfbhCFCMFlGq5xFEKakMHganEEmdFBVYNFJLF4okrRpdHpFF5ohjFDgdKJHC43EnUptcFkAlSNFQ2ZwXlIXcqMlQ9FEaItFH8MAY64peN6DgGg5x4Wpw7TPFG8bGllQE54k9WqZ5FKtqYo9PxXBDMjTOl8fn0KDgBi9cPIlKp1FodH9uNnjNjzMO0ChcLhltuMMo5KY5gYAAvBgAAGPoYBQgZJJMWhCMBiEYNylTAJBvqaFMYBJAEbbePWAT7L6AC+uAQTgP6xFC6LiCBXCkI6ABiNFXEk-6AaQ+xfjgKAoBgjGwvCGAam4dC4LxGAACo0pk7YRJYOaDA8lx8HQxInpalKZjSdJYmB0HJHBdYYMwplgRgGhnqQuDUaCSSVH61w8XxADKqo4OqkqieJfHEJiFCWOoHzHqSmlUjQ2jUrSTpgakbzCOZlncFyPFoLgRQ6GhDBgLZ4j2bgxG6PMM5JJxWW+qY7hUJ0zBTGcehJFgIAALJFky7ZQQgYlwMIDmQXFVCOpwvojaNY3jQVuAfrgcAgOIyCzZ4UCgHoyAaEYFgQMRxFAA)

TL;DR; 

Executing a dynamically created code while keeping your current app working involves using `try/catch` block, which prevents debugger (Chromium's at least) from hit to the correct exception point. 

Both including the sourcemaps (`embedded` option worked for me) and enabling **Pause on exceptions** in debugger window (see [this](https://github.com/ceremcem/aecad/issues/8#issuecomment-439483506)) makes the debugger hit the exact exception line.
